entities/vectorindex/hnsw/config.go

```diff
@@ -109,82 +109,91 @@ func ParseAndValidateConfig(input interface{}) (schema.VectorIndexConfig, error)
 
 	asMap, ok := input.(map[string]interface{})
 	if !ok || asMap == nil {
-		return uc, fmt.Errorf("input must be a non-nil map")
+		return nil, fmt.Errorf("input must be a non-nil map")
```

Comment:
why in all of those places the `return uc, ...` has been changed to `return nil, ...`?

Reply:
This can obviously be up to debate, but I learned that functions should generally return nil as a value if an error is returned. I did not see why this should be different here.